src/stave.js

```diff
@@ -319,14 +298,96 @@ Vex.Flow.Stave = (function() {
 
     addEndTimeSignature: function(timeSpec, customPadding) {
       this.addEndModifier(new Vex.Flow.TimeSignature(timeSpec, customPadding));
+      return this;
     },
 
     addTrebleGlyph: function() {
-      this.clef = "treble";
-      this.addGlyph(new Vex.Flow.Glyph("v83", 40));
+      this.addClef('treble');
       return this;
     },
 
+    getModifiers: function(position) {
+      if (position === undefined) return this.modifiers;
+
+      return this.modifiers.filter(function(modifier) {
+        return modifier.getPosition() === position;
+      });
+    },
+
+    sortByCategory: function(items, order) {
+      for (var i = items.length - 1; i >= 0; i--) {
+        for (var j = 0; j < i; j++) {
+          if (order[items[j].getCategory()] > order[items[j + 1].getCategory()]) {
+            var temp = items[j];
+            items[j] = items[j + 1];
+            items[j + 1] = temp;
+          }
+        }
+      }
+    },
+
+    format: function() {
```

Comment:
I just added sortByCategory unit test to stave_tests.

And today I implemented stave modifier set functions, I didn't push yet;
setClef, setEndClef, setKeySignature, setEndKeySignature, setTimeSignature and setEndTimeSignature.

We can keep both add and set. It will not also break any existing tests and no effect to existing users.
Another thing I'd like to get your opinion is about updating current test cases.
I think it's better to replace all current addClef and etc add functions test cases to use set functions except some test cases which add multiple stave modifiers into a single stave.
How do you think of that?
Both set and add are ok to use but in usual cases I think set functions fit more.